src/features/island/buildings/components/building/BuildingOilTank.tsx

```diff
@@ -0,0 +1,325 @@
+import oilBarrel from "assets/icons/oil_barrel.webp";
+import { ITEM_DETAILS } from "features/game/types/images";
+import { useAppTranslation } from "lib/i18n/useAppTranslations";
+import {
+  BuildingName,
+  CookingBuildingName,
+} from "features/game/types/buildings";
+import { BUILDING_DAILY_OIL_CAPACITY } from "features/game/events/landExpansion/supplyCookingOil";
+import {
+  BUILDING_DAILY_OIL_CONSUMPTION,
+  BUILDING_OIL_BOOSTS,
+  isCookingBuilding,
+} from "features/game/events/landExpansion/cook";
+import { COOKABLES, CookableName } from "features/game/types/consumables";
+import { Context } from "features/game/GameProvider";
+import { ModalOverlay } from "components/ui/ModalOverlay";
+import { InnerPanel } from "components/ui/Panel";
+import { SUNNYSIDE } from "assets/sunnyside";
+import { setPrecision } from "lib/utils/formatNumber";
+import Decimal from "decimal.js-light";
+import { Box } from "components/ui/Box";
+import useUiRefresher from "lib/utils/hooks/useUiRefresher";
+
+interface OilTankProps {
+  currentlyCooking: CookableName | undefined;
+  buildingName: BuildingName;
+  buildingId: string;
+}
+
+const OIL_INCREMENT_AMOUNT = 1;
+
+export const BuildingOilTank = ({
+  currentlyCooking,
+  buildingName,
+  buildingId,
+}: OilTankProps) => {
+  const { gameService } = useContext(Context);
+  const [showAddOilModal, setShowAddOilModal] = useState<boolean>(false);
+  const { t } = useAppTranslation();
+  const game = gameService.state.context.state;
+
+  const [totalOilToAdd, setTotalOilToAdd] = useState(0);
+
+  const building = game.buildings[buildingName]?.find(
+    (building) => building.id === buildingId
+  );
+
+  const oilRemainingInBuilding = building?.oilRemaining || 0;
+
+  const incrementOil = () => {
+    setTotalOilToAdd((prev) => prev + OIL_INCREMENT_AMOUNT);
+  };
+
+  const decrementOil = () => {
+    setTotalOilToAdd((prev) => Math.max(prev - OIL_INCREMENT_AMOUNT, 0));
+  };
+
+  function getOilTimeInMillis(oil: number): number {
+    if (!isCookingBuilding(buildingName)) {
+      return 0;
+    }
+
+    const dailyOilConsumption = BUILDING_DAILY_OIL_CONSUMPTION[buildingName];
+    const hoursPerDay = 24;
+    const oilTimeInHours = (oil / dailyOilConsumption) * hoursPerDay;
+
+    // Convert hours to milliseconds
+    const millisecondsPerHour = 60 * 60 * 1000;
+    const oilTimeInMillis = oilTimeInHours * millisecondsPerHour;
+
+    return oilTimeInMillis;
+  }
+
+  const handleSupplyOil = (amount: number) => {
+    gameService.send("cookingOil.supplied", {
+      building: buildingName,
+      buildingId,
+      oilQuantity: amount,
+    });
+
+    gameService.send("SAVE");
+
+    setShowAddOilModal(false);
+  };
+
+  const canAddOil = () => {
+    if (!isCookingBuilding(buildingName)) {
+      return false;
+    }
+
+    return (
+      oilRemainingInBuilding <= BUILDING_DAILY_OIL_CAPACITY[buildingName] - 1
+    );
+  };
+
+  const canIncrementOil = () => {
```

Comment:
<img width="431" alt="Screenshot 2024-05-28 at 3 01 59 PM" src="https://github.com/sunflower-land/sunflower-land/assets/11745561/15c0f190-6d10-4e43-bb45-e405291fa61a">

It looks like the UI supports adding "+1" oil even when I didn't have any oil (my inventory was empty)